### ISSUE TITLE
vertx 4.0.2

### ISSUE THREAD

**Issue by sheldon-markk** [vertx-howtos/graal-native-image-howto#8]
We are trying to get this to work with vertx 4.x but Netty internal packages seems to be causing too many issues

```
org.slf4j.impl.StaticLoggerBinder was unintentionally initialized at build time. io.netty.util.internal.SystemPropertyUtil caused initialization of this class with the following trace: 
	at org.slf4j.impl.StaticLoggerBinder.<clinit>(StaticLoggerBinder.java:45)
	at org.slf4j.LoggerFactory.bind(LoggerFactory.java:150)
	at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:124)
	at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:412)
	at io.netty.util.internal.logging.Slf4JLoggerFactory.<init>(Slf4JLoggerFactory.java:42)
	at io.netty.util.internal.logging.InternalLoggerFactory.newDefaultFactory(InternalLoggerFactory.java:44)
	at io.netty.util.internal.logging.InternalLoggerFactory.getDefaultFactory(InternalLoggerFactory.java:69)
	at io.netty.util.internal.logging.InternalLoggerFactory.getInstance(InternalLoggerFactory.java:92)
	at io.netty.util.internal.logging.InternalLoggerFactory.getInstance(InternalLoggerFactory.java:85)
	at io.netty.util.internal.SystemPropertyUtil.<clinit>(SystemPropertyUtil.java:29)

io.netty.util.internal.InternalThreadLocalMap the class was requested to be initialized at run time (from the command line). io.netty.handler.codec.DateFormatter caused initialization of this class with the following trace: 
	at io.netty.util.internal.InternalThreadLocalMap.<clinit>(InternalThreadLocalMap.java:41)
	at io.netty.util.concurrent.FastThreadLocal.<clinit>(FastThreadLocal.java:46)
	at io.netty.handler.codec.DateFormatter.<clinit>(DateFormatter.java:73)

org.apache.log4j.DailyRollingFileAppender was unintentionally initialized at build time. io.netty.util.internal.SystemPropertyUtil caused initialization of this class with the following trace: 
	at org.apache.log4j.DailyRollingFileAppender.<clinit>(DailyRollingFileAppender.java:187)
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Unknown Source)
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.lang.Class.newInstance(Class.java:584)
	at org.apache.log4j.helpers.OptionConverter.instantiateByClassName(OptionConverter.java:337)
	at org.apache.log4j.helpers.OptionConverter.instantiateByKey(OptionConverter.java:124)
	at org.apache.log4j.PropertyConfigurator.parseAppender(PropertyConfigurator.java:785)
	at org.apache.log4j.PropertyConfigurator.parseCategory(PropertyConfigurator.java:768)
	at org.apache.log4j.PropertyConfigurator.configureRootCategory(PropertyConfigurator.java:648)
	at org.apache.log4j.PropertyConfigurator.doConfigure(PropertyConfigurator.java:514)
	at org.apache.log4j.PropertyConfigurator.doConfigure(PropertyConfigurator.java:580)
	at org.apache.log4j.helpers.OptionConverter.selectAndConfigure(OptionConverter.java:526)
	at org.apache.log4j.LogManager.<clinit>(LogManager.java:127)
	at org.slf4j.impl.Log4jLoggerFactory.<init>(Log4jLoggerFactory.java:66)
	at org.slf4j.impl.StaticLoggerBinder.<init>(StaticLoggerBinder.java:72)
	at org.slf4j.impl.StaticLoggerBinder.<clinit>(StaticLoggerBinder.java:45)
	at org.slf4j.LoggerFactory.bind(LoggerFactory.java:150)
	at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:124)
	at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:412)
	at io.netty.util.internal.logging.Slf4JLoggerFactory.<init>(Slf4JLoggerFactory.java:42)
	at io.netty.util.internal.logging.InternalLoggerFactory.newDefaultFactory(InternalLoggerFactory.java:44)
	at io.netty.util.internal.logging.InternalLoggerFactory.getDefaultFactory(InternalLoggerFactory.java:69)
	at io.netty.util.internal.logging.InternalLoggerFactory.getInstance(InternalLoggerFactory.java:92)
	at io.netty.util.internal.logging.InternalLoggerFactory.getInstance(InternalLoggerFactory.java:85)
	at io.netty.util.internal.SystemPropertyUtil.<clinit>(SystemPropertyUtil.java:29)
```

**Comment by sheldon-markk** [comment]
GraalVM 21.0.0.2 used here.

**Comment by ksclarke** [comment]
This might be fixed by #10 which added support for 4.2.3.

**Comment by pmlopes** [comment]
It's hard to maintain many versions, so let's assume the current 4.2.3 support is working and reopen if really needed.
